packages/rum/src/getActionNameFromElement.ts

```diff
@@ -0,0 +1,176 @@
+export function getActionNameFromElement(element: Element): string {
+  return (
+    getActionNameFromElementForGetters(element, priorityGetters) ||
+    getActionNameFromElementForGetters(element, fallbackGetters) ||
+    ''
+  )
+}
+
+type NameGetter = (element: Element | HTMLElement | HTMLInputElement | HTMLSelectElement) => string | undefined | null
+
+const priorityGetters: NameGetter[] = [
+  // associated LABEL text
+  (element) => {
+    // IE does not support element.labels, so we fallback to a CSS selector based on the element id
+    // instead
+    if (supportsLabelProperty()) {
+      if ('labels' in element && element.labels && element.labels.length > 0) {
+        return getTextualContent(element.labels[0])
+      }
+    } else if (element.id) {
+      const label =
+        element.ownerDocument && element.ownerDocument.querySelector(`label[for="${element.id.replace('"', '\\"')}"]`)
+      return label && getTextualContent(label)
+    }
+  },
+  // INPUT button (and associated) value
+  (element) => {
+    if (element.nodeName === 'INPUT') {
+      const input = element as HTMLInputElement
+      const type = input.getAttribute('type')
+      if (type === 'button' || type === 'submit' || type === 'reset') {
+        return input.value
+      }
+    }
+  },
+  // BUTTON, LABEL or button-like element text
+  (element) => {
+    if (element.nodeName === 'BUTTON' || element.nodeName === 'LABEL' || element.getAttribute('role') === 'button') {
+      return getTextualContent(element)
+    }
+  },
+  // aria-label attribute value
+  (element) => element.getAttribute('aria-label'),
+  // associated element text designated by the aria-labelledby attribute
+  (element) => {
+    const labelledByAttribute = element.getAttribute('aria-labelledby')
+    if (labelledByAttribute) {
+      return labelledByAttribute
+        .split(/\s+/)
+        .map((id) => getElementById(element, id))
+        .filter((label): label is HTMLElement => Boolean(label))
+        .map(getTextualContent)
+        .join(' ')
+    }
+  },
+  // alt attribute value
+  (element) => element.getAttribute('alt'),
+  // name attribute value
+  (element) => element.getAttribute('name'),
+  // title attribute value
+  (element) => element.getAttribute('title'),
+  // placeholder attribute value
+  (element) => element.getAttribute('placeholder'),
+  // SELECT first OPTION text
+  (element) => {
```

Comment:
I thought this was nice to have comments for every getters so it's easier to scan. But you are right, attributes getters are very trivial so it may not be really needed. I can remove those if you think it's too much.

Reply:
I think this kind of comments can be useful to separate chunks of related behaviors in a given context.
Here, the functions are quite simple so IMO it adds more noise than it helps the understanding.